cloto/manager/AuthorizationManager.py

```diff
@@ -36,47 +39,96 @@ class AuthorizationManager():
     """
     myClient = None
     client = requests
+    session = None
 
-    def generate_adminToken(self, username, password, url):
-        """This method generates an admin token."""
+    def generate_session(self, username, password, tenant_id, tenant_name, user_domain_name, auth_api, url):
+        """
+        Init the variables related to authorization, needed to execute tests
+        :return: The auth token retrieved
+        """
+
+        cred_kwargs = {
+            'auth_url': url,
+            'username': username,
+            'password': password
+        }
+
+        # Currently, both v2 and v3 Identity API versions are supported
+        if auth_api == 'v2.0':
+            cred_kwargs['tenant_name'] = tenant_name
+        elif auth_api == 'v3':
+            cred_kwargs['user_domain_name'] = user_domain_name
+
```

Comment:
Given that 'tenant_name' and 'user_domain_name' are optional (depending on V2 or V3), do not include them as mandatory function arguments. You could use **kwargs instead.